### ISSUE TITLE
Cannot install CCXT

### ISSUE THREAD

**Issue by hiteshbhatia3559** [ccxt/ccxt#3617]
ATTENTION!!!

MUST READ THIS BEFORE SUBMITTING ISSUES:

https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#how-to-submit-an-issue

- OS: Windows 10 Enterprise
- Programming Language version: Python 3.6.5
- CCXT version: Latest from pip
- Exchange: -
- Method: -

Hi,

Having the following issue. Already tried to install VC++ 2014 through 17. Couldn't install Visual Studio 2017 because I have a data cap - however I didn't know VS would be a dependency.

Console log as below

```
`Requirement already satisfied: six in c:\users\hitesh.bhatia\appdata\local\continuum\anaconda3\lib\site-packages (from attrdict<3,>=2.0.0->eth-account<0.3.0,>=0.2.1->web3==4.4.1; python_version >= "3.5.2"->ccxt) (1.11.0)
Building wheels for collected packages: lru-dict
  Running setup.py bdist_wheel for lru-dict ... error
  Complete output from command c:\users\hitesh.bhatia\appdata\local\continuum\anaconda3\python.exe -u -c "import setuptools, tokenize;__file__='C:\\Users\\HITESH1.BHA\\AppData\\Local\\Temp\\pip-install-weqh1ncy\\lru-dict\\setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d C:\Users\HITESH1.BHA\AppData\Local\Temp\pip-wheel-jid_c4sy --python-tag cp36:
  running bdist_wheel
  running build
  running build_ext
  building 'lru' extension
  error: Microsoft Visual C++ 14.0 is required. Get it with "Microsoft Visual C++ Build Tools": http://landinghub.visualstudio.com/visual-cpp-build-tools

  ----------------------------------------
  Failed building wheel for lru-dict
  Running setup.py clean for lru-dict
Failed to build lru-dict
Installing collected packages: lru-dict, web3, ccxt
  Running setup.py install for lru-dict ... error
    Complete output from command c:\users\hitesh.bhatia\appdata\local\continuum\anaconda3\python.exe -u -c "import setuptools, tokenize;__file__='C:\\Users\\HITESH1.BHA\\AppData\\Local\\Temp\\pip-install-weqh1ncy\\lru-dict\\setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record C:\Users\HITESH1.BHA\AppData\Local\Temp\pip-record-6n5cpz6a\install-record.txt --single-version-externally-managed --compile:
    running install
    running build
    running build_ext
    building 'lru' extension
    error: Microsoft Visual C++ 14.0 is required. Get it with "Microsoft Visual C++ Build Tools": http://landinghub.visualstudio.com/visual-cpp-build-tools

    ----------------------------------------
Command "c:\users\hitesh.bhatia\appdata\local\continuum\anaconda3\python.exe -u -c "import setuptools, tokenize;__file__='C:\\Users\\HITESH1.BHA\\AppData\\Local\\Temp\\pip-install-weqh1ncy\\lru-dict\\setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record C:\Users\HITESH1.BHA\AppData\Local\Temp\pip-record-6n5cpz6a\install-record.txt --single-version-externally-managed --compile" failed with error code 1 in C:\Users\HITESH~1.BHA\AppData\Local\Temp\pip-install-weqh1ncy\lru-dict\`
```

**Comment by hiteshbhatia3559** [comment]
Hi, which question is this a duplicate of?

**Comment by kroitor** [comment]
@hiteshbhatia3559 it says:

> Microsoft Visual C++ 14.0 is required. Get it with "Microsoft Visual C++ Build Tools": http://landinghub.visualstudio.com/visual-cpp-build-tools

Downloading the Build Tools from the above link, then following install instructions should solve the issue for you. Those are Build Tools, not the MSVC itself. Let us know if it does not help.

**Comment by kroitor** [comment]
> Hi, which question is this a duplicate of?

Duplicate: https://github.com/ccxt/ccxt/issues/3578

**Comment by hiteshbhatia3559** [comment]
If you have this issue, Install Visual Studio 2017. Nothing else fixed it for me, this did.

**Comment by kroitor** [comment]
@hiteshbhatia3559 thx for reporting back!

**Comment by Tehsurfer** [comment]
Hi all, just checking in to ask if any of you have been able to get the build tools on Ubuntu? Having a bit of trouble getting ccxt to run on a server.

**Comment by kroitor** [comment]
@Tehsurfer Hey!

> just checking in to ask if any of you have been able to get the build tools on Ubuntu?

Yep, we're using it on servers running with Ubuntu and other OSes...

> Having a bit of trouble getting ccxt to run on a server.

Ok, but, what's the particular trouble you're having? (be as specific and precise as you possibly can: https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#how-to-submit-an-issue)

**Comment by Tehsurfer** [comment]
@kroitor Hi again :) Thanks for following up on my question. Funnily enough I just went back to see what I had tried and it is somehow working now

For those that are also having issues here was what worked for me:

 1. I had to switch from python 3.5.1 to 3.6 (Ubuntu 16.04.5 LTS server) 

2. `curl https://packages.microsoft.com/keys/microsoft.asc | gpg --dearmor > microsoft.gpg`

3. `sudo sh -c 'echo "deb [arch=amd64] https://packages.microsoft.com/repos/vscode stable main" > /etc/apt/sources.list.d/vscode.list'`

3.b `sudo add-apt-repository -y "deb [arch=amd64] https://packages.microsoft.com/repos/vscode stable main"`

4. `pip3 install setuptools`

5.  rename 'apt_pkg.cpython-35m-x86_64-linux-gnu.so' to 'apt_pkg.so'  

6. reinstall cytoolz

pip3 install ccxt works for me after the above although I am not entirely sure what exactly I have done.
